yuyo/components.py

```diff
@@ -2714,17 +3271,49 @@ def __init__(self, special_string: str, timeout: typing.Optional[datetime.timede
     )
     ```
 
-    !!! note
-        Since decorators are executed from the bottom upwards fields added
-        through decorator calls will follow the same order.
+    There's also class descriptors which can be used to declare static
+    components. The following descriptors work by decorating their
+    component's callback:
+
+    * [as_button][yuyo.components.as_button]
+    * [as_select_menu][yuyo.components.as_select_menu]
+    * [as_channel_select][yuyo.components.as_channel_select]
+    * [as_text_select][yuyo.components.as_text_select]
+
+    [link_button][yuyo.components.link_button] returns a descriptor without
+    decorating any callback.
+
+    ```py
+    class CustomColumn(components.ActionColumnExecutor):
+        @components.as_button(ButtonStyle.PRIMARY, label="label")
+        async def left_button(self, ctx: components.ComponentContext) -> None:
+            ...
+
+        link_button = components.link_button(url="https://example.com", label="Go to page")
+
+        @components.as_button(ButtonStyle.SECONDARY, label="meow")
+        async def right_button(self, ctx: components.ComponentContext) -> None:
+            ...
+
+        @components.as_channel_select(channel_types=[hikari.TextableChannel])
+        async def text_select_menu(self, ctx: components.ComponentContext) -> None:
+            ...
+    ```
     """
 
     __slots__ = ("_last_triggered", "_rows", "_timeout")
 
     _all_static_rows: typing.ClassVar[list[ActionRowExecutor]] = []
-    _static_fields: typing.ClassVar[
-        list[tuple[hikari.ComponentType, collections.Callable[[ActionRowExecutor], object]]]
-    ] = []
+    """Sequence of all the static rows on this class.
+
+    This includes inherited fields.
+    """
+
+    _static_fields: typing.ClassVar[list[collections.Callable[[type[Self]], object]]] = []
```

Comment:
TODO: consider storing these as `dict[custom_id, Callable]` instead of using `magic=True`. This might allow for preferable behaviour around overriding existing components